### PR TITLE
test conclusion, session metadata, sessionError

### PR DESCRIPTION
**Pull request by boazin**

### CODE REVIEW

**Comment by vmalloc** [comment]
Travis failed....

**Comment by vmalloc** [comment]
Oh sorry - just noticed it's because of the backslash-python dep.

File: flask_app/models.py [hunk 1]
```diff
@@ -95,11 +108,26 @@ class TestMetadata(db.Model):
...
+class TestError(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
     test_id = db.Column(db.Integer, db.ForeignKey('test.id', ondelete='CASCADE'), index=True)
     traceback = db.Column(JSON)
     exception_type = db.Column(db.String(256), index=True)
     exception = db.Column(db.String(256), index=True)
+    timestamp = db.Column(db.Float, default=get_current_time)
+
+
+class SessionError(db.Model):
```
Comment (vmalloc) [inline flask_app/models.py]: These two should be a single table (Error) and two many-to-many tables linking errors to sessions and tests respectively. Code duplication here will likely get out of sync...

Comment (boazin) [inline flask_app/models.py]: That's what I was talking about in the email the other day, it requires
polymorphism (test_id, session_id - in that table)
Which is the "correct" approach but in SQLAlchemy means tons of code

Comment (vmalloc) [inline flask_app/models.py]: So perhaps I misunderstood the question and I apologize. 

Why does this require polymorphism? At no point is there a need to make the error behave like two different errors. We always approach the error from some route (be it test or session, it doesn't matter) and we render it the same way... Why wouldn't a simple pair of many-to-many tables suffice here?

**Comment by boazin** [comment]
Updated the PR. Travis fails due to backslash-python non-compatability

**Pull request merged**